kotlinx-coroutines-core/common/src/JobSupport.kt

```diff
@@ -257,6 +262,8 @@ public open class JobSupport constructor(active: Boolean) : Job, ChildJob, Paren
         check(state is Empty || state is JobNode<*>) // only simple state without lists where children can concurrently add
         check(update !is CompletedExceptionally) // only for normal completion
         if (!_state.compareAndSet(state, update.boxIncomplete())) return false
+        onCancellation(null) // simple state is not a failure
```

Comment:
If it is not, then maybe we should not invoke `onCancellation`?
Especially when its doc says "This function is invoked once when this job is being cancelled"

Reply:
It is now used only in `actor`, so there are no bugs, but nevertheless, we contradict our own contract

Reply:
Good catch. I've updated it to be be invoked only on cancellation and also renamed it to `onCancelling` for consistency with `invokeOnCompletion(onCancelling=xxx)` parameter. Also, it turns out that the condition it was called under from `tryFinalizeFinishingState` was not correct, but it did not really matter (since it is used only from `actor` extra calls from there do not harm anyway).